specs/plugin-extensibility.md

```diff
@@ -20,15 +20,41 @@ Keys and associated certificates used for signing artifacts using Notation could
   * Notation MUST work with a plugin that implements a matching or lower minor version of the plugin contract. Notation SHALL NOT support using a plugin with higher version of plugin contract.
   * A plugin MUST support a single plugin contract version, per major version.
 
-Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads. This mechanism is used as Go language (used to develop [Notation Go library](https://github.com/notaryproject/notation-go)) does not have a [in built support](https://github.com/golang/go/issues/19282) to load and execute plugins that works across OS platforms. Other mechanisms like gRPC require every plugin to be implemented as a service/daemon.
+Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads.
+
+### Package a plugin
+
+Plugin publisher SHOULD follow these conventions to package and distribute plugin files:
+
+* The file formats supported by Notation include `.zip`, `.tar.gz`, and single plugin executable file. It is highly recommended to package the single plugin executable file into an archive for installation security and efficiency consideration
+* Plugin executable file MUST follow the naming convention `notation-{plugin-name}` 
+* Notation supports installing a plugin from an `https` URL or from a file in the filesystem
+* If the archive format is `.zip` or `.tar.gz`, there MUST be one and only one plugin executable file within each archive
+* Plugin publisher SHOULD provide SHA256 checksum for each released archive 
+* If the plugin has external dependencies, plugin publisher MUST include them in the archive
+* License file is optional but it's recommended to be included in the archive
+* To avoid [ZIP bombs](https://en.wikipedia.org/wiki/Zip_bomb), plugin archive file size is limited to less than 256 MiB. The sum of the extracted files is also limited to 256 MiB
+
+For example, an archive of a Notation plugin `helloworld` for Linux AMD64 machine `notation-helloworld_1.0.1_linux_amd64.zip` includes these files:
+
+```
+notation-helloworld_1.0.1_linux_amd64
+├── notation-helloworld (must)
+├── LICENSE (optional)
+└── Dependency file (optional)
+```
```

Comment:
```suggestion
### Packaging Plugin for Distribution

Notation supports the installation of a plugin from an `https` URL or from a file in the filesystem. To support  the installation of plugin using `notation plugin install` commands, the plugin distributor must adhere to the following conventions:

1. Plugin executable file MUST follow the naming convention `notation-{plugin-name}`
1. The plugin distribution format MUST be either .zip, .tar.gz, or a single plugin executable file.
1. If archive format is .zip or .tar.gz, there MUST be exactly one plugin executable file within the archive.
1. If the plugin has external dependencies, plugin publisher SHOULD include them in the archive
1. The plugin archive MAY contain License files. Also, it's recommended to include licenses in the archive.
1. Currently, Notation facilitates the installation of plugin executable files and archives, with a size limit of less than 256 MiB. Consequently, the plugin executable file and archive size MUST be less than 256 MiB. Note: Although users have the option to install plugins larger than 256 MiB, they will be unable to utilize the `notation plugin install` command in such cases.

For example, consider an archive of a Notation plugin  named `helloworld` developed for Linux AMD64 machine `notation-helloworld_1.0.1_linux_amd64.zip` includes following files:
```shell
notation-helloworld_1.0.1_linux_amd64
├── notation-helloworld (must)
├── LICENSE             (optional)
└── dependencies        (optional)
```
```
* Zips are more vulnerable than executable binary (more complex, more attack vectors)
* Ideally for distribution, package (either archive or executable) must be signed (authenticity and integrity) but currently notation supports only integrity validation.

Reply:
Signing and verifying a plugin relies on the arbitrary file signing and verification support. We can add this until notation supports it in the next release.